src/tiled/variantpropertymanager.cpp

```diff
@@ -126,6 +135,32 @@ QString VariantPropertyManager::valueText(const QtProperty *property) const
         QVariant value = mValues[property].value;
         int typeId = propertyType(property);
 
+        if (typeId == objectIdTypeId()) {
+            ObjectId id = value.value<ObjectId>();
+
+            if (MapDocument *document = DocumentManager::instance()->currentDocument()) {
```

Comment:
Hmm, no, you'd need a `Tileset*` as well. I think adding a `Map*` and a `Tile*` would work.